pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1135,19 +1136,22 @@ public CompletableFuture<Optional<Topic>> getTopic(final TopicName topicName, bo
                         }
                         topicEventsDispatcher.notify(topicName.toString(), TopicEvent.LOAD, EventStage.FAILURE);
                         return CompletableFuture.completedFuture(Optional.empty());
-                    });
+                    }, pulsar.getExecutor());
                 } else if (createIfMissing) {
-                    return topics.computeIfAbsent(topicName.toString(), (name) -> {
-                        topicEventsDispatcher.notify(topicName.toString(), TopicEvent.CREATE, EventStage.BEFORE);
-
-                        CompletableFuture<Optional<Topic>> res = createNonPersistentTopic(name);
-
-                        CompletableFuture<Optional<Topic>> eventFuture = topicEventsDispatcher
-                                .notifyOnCompletion(res, topicName.toString(), TopicEvent.CREATE);
-                        topicEventsDispatcher
-                                .notifyOnCompletion(eventFuture, topicName.toString(), TopicEvent.LOAD);
-                        return res;
-                    });
+                    CompletableFuture<Void> createTopicFuture = new CompletableFuture<>();
+                    return createTopicFuture.thenComposeAsync(__ -> {
```

Comment:
What is the purpose of this? I don't currently see from the code how `createTopicFuture` would get completed. (I'll have to take a closer look at the code to understand it)

Reply:
> What is the purpose of this? I don't currently see from the code how createTopicFuture would get completed. (I'll have to take a closer look at the code to understand it)

It is unnecessary, I have removed these changes